cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -1197,4 +1198,84 @@ private String appendScopeAndKeyToUrl(String workflowTokenUrl, @Nullable Workflo
     }
     return output.toString();
   }
+
+  private String createInputForRecordVerification(String folderName) throws IOException {
+    File inputDir = tmpFolder.newFolder(folderName);
+
+    File inputFile = new File(inputDir.getPath() + "/words.txt");
+    try (BufferedWriter writer = Files.newBufferedWriter(inputFile.toPath(), Charsets.UTF_8)) {
+      writer.write("id1:value1");
+      writer.newLine();
+      writer.write("id2:value2");
+      writer.newLine();
+      writer.write("id3:value3");
+    }
+    return inputDir.getAbsolutePath();
+  }
+
+  @Test
+  public void testWorkflowPutOperation() throws Exception {
+    Assert.assertEquals(200, deploy(WorkflowTokenTestPutApp.class).getStatusLine().getStatusCode());
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, WorkflowTokenTestPutApp.NAME);
+    Id.Workflow workflowId = Id.Workflow.from(appId, WorkflowTokenTestPutApp.WorkflowTokenTestPut.NAME);
+
+    // Start program with "put.in.mapper.initialize" argument.
+    // It will perform put operation on the WorkflowToken in the Initialize method of the Mapper class.
+    // This should fail.
+    String outputPath = new File(tmpFolder.newFolder(), "output").getAbsolutePath();
+    startProgram(workflowId, ImmutableMap.of("inputPath", createInputForRecordVerification("firstInput"),
+                                             "outputPath", outputPath, "put.in.mapper.initialize", "true"));
+    waitState(workflowId, ProgramRunStatus.RUNNING.name());
+    waitState(workflowId, "STOPPED");
+
+    List<RunRecord> programRuns = getProgramRuns(workflowId, ProgramRunStatus.FAILED.name());
```

Comment:
`waitState` queries the `/status` api, which checks for the program in the in-memory runtimeInfo object. If the program is not found, it returns `STOPPED`, otherwise it returns `RUNNING`. When the program completes/fails, it is removed from the runTimeInfo and in the `RunRecord` we mark the appropriate status as COMPLETED/FAILED etc.